File: go/crawler/crawler.go
package main

import (
	"fmt"
	"net/http"
	"io/ioutil"
	"strings"
	"time"
	"regexp"
	"strconv"
)

func fetch(url string) string{
	fmt.Println("Fetch Url", url)
	client := &http.Client{}
	req, _ := http.NewRequest("GET", url, nil)

	req.Header.Set("User-Agent", "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.190 Safari/537.36")
	resp, err := client.Do(req)
	if err != nil{
		fmt.Println("Http Get error", err)
	}
	if resp.StatusCode != 200 {
		fmt.Println("Http StatusCode ", resp.StatusCode)
	}

	defer resp.Body.Close()
	body, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		fmt.Println("Read error", err)
	}
	return string(body)
}

func parseUrls(url string){
	body := fetch(url)

	body = strings.Replace(body, "\n", "", -1)

	rp := regexp.MustCompile(`<div class="hd">(.*?)</div>`)
	titleRe := regexp.MustCompile(`<span class="title">(.*?)</span>`)
	idRe := regexp.MustCompile(`<a href="https://movie.douban.com/subject/(\d+)/"`)

	items := rp.FindAllStringSubmatch(body, -1)
	for _, item := range items {
		fmt.Println(idRe.FindStringSubmatch(item[1])[1], titleRe.FindStringSubmatch(item[1])[1])
	}
}
func main(){
	start := time.Now()
	
	for i := 0; i < 10; i++ {
		parseUrls("https://movie.douban.com/top250?start=" + strconv.Itoa(25*i))
	}

	elapsed := time.Since(start)
	fmt.Printf("Took %s\n", elapsed)
}

File: python/crawler/crawler.py
from bs4 import BeautifulSoup
from lxml import html
import xml
import requests
import pyppeteer

url = "https://movie.douban.com/chart"
headers = {'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.141 Safari/537.36'}
f = requests.get(url, headers=headers)			# Get该网页从而获得该HTML内容
print(f.status_code)
soup = BeautifulSoup(f.content, 'lxml')		# 用lxml解析器解析该网页的内容， 好像f.txt也是返回的html
# print(f.content.decode())		# 尝试打印出网页内容，看是否获取成功
# content = soup.find_all('div', class_="p12")	# 尝试获取节点，因为class和关键字冲突，所以改名class_

print(soup.title)
print(soup.title.text)

# for k in soup.find_all('div', class_='name'):
# 	print(k.find_all('a').string)
